diff --git a/src/arch/x86/isa/insts/x87/arithmetic/subtraction.py b/src/arch/x86/isa/insts/x87/arithmetic/subtraction.py
index 2275757..09abfb9 100644
--- a/src/arch/x86/isa/insts/x87/arithmetic/subtraction.py
+++ b/src/arch/x86/isa/insts/x87/arithmetic/subtraction.py
@@ -73,6 +73,43 @@ def macroop FSUB2_P
     subfp st(0), st(0), ufp1
 };
 
+def macroop FSUBR1_R
+{
+    subfp st(0), sti, st(0)
+};
+
+
+def macroop FSUBR1_M
+{
+    ldfp ufp1, seg, sib, disp
+    subfp st(0), ufp1, st(0)
+};
+
+def macroop FSUBR1_P
+{
+    rdip t7
+    ldfp ufp1, seg, riprel, disp
+    subfp st(0), ufp1, st(0)
+};
+
+def macroop FSUBR2_R
+{
+    subfp sti, st(0), sti
+};
+
+def macroop FSUBR2_M
+{
+    ldfp ufp1, seg, sib, disp
+    subfp st(0), ufp1, st(0)
+};
+
+def macroop FSUBR2_P
+{
+    rdip t7
+    ldfp ufp1, seg, riprel, disp
+    subfp st(0), ufp1, st(0)
+};
+
 def macroop FSUBP
 {
     subfp st(1), st(1), st(0), spm=1
